timelock-agent/src/main/java/com/palantir/atlasdb/timelock/paxos/BatchingNetworkClientFactories.java

```diff
@@ -31,32 +36,46 @@
     abstract int quorumSize();
     abstract ExecutorService sharedExecutor();
 
-    public NetworkClientFactories factories() {
+    @Value.Auxiliary
+    @Value.Derived
+    AutobatchingPaxosAcceptorNetworkClientFactory acceptorNetworkClientFactory() {
         List<BatchPaxosAcceptor> allBatchAcceptors = metrics()
                 .instrumentLocalAndRemotesFor(
                         BatchPaxosAcceptor.class,
                         components().batchAcceptor(),
                         UseCaseAwareBatchPaxosAcceptorAdapter.wrap(useCase(), remoteClients().batchAcceptor()),
                         "batch-paxos-acceptor")
                 .all();
+        return AutobatchingPaxosAcceptorNetworkClientFactory.create(allBatchAcceptors, sharedExecutor(), quorumSize());
+    }
 
-        AutobatchingPaxosAcceptorNetworkClientFactory acceptorFactory =
-                AutobatchingPaxosAcceptorNetworkClientFactory.create(allBatchAcceptors, sharedExecutor(), quorumSize());
-
+    @Value.Auxiliary
+    @Value.Derived
+    AutobatchingPaxosLearnerNetworkClientFactory learnerNetworkClientFactory() {
         LocalAndRemotes<BatchPaxosLearner> allBatchLearners = metrics()
                 .instrumentLocalAndRemotesFor(
                         BatchPaxosLearner.class,
                         components().batchLearner(),
                         UseCaseAwareBatchPaxosLearnerAdapter.wrap(useCase(), remoteClients().batchLearner()),
                         "batch-paxos-learner");
 
-        AutobatchingPaxosLearnerNetworkClientFactory learnerFactory =
-                AutobatchingPaxosLearnerNetworkClientFactory.create(allBatchLearners, sharedExecutor(), quorumSize());
+        return AutobatchingPaxosLearnerNetworkClientFactory.create(allBatchLearners, sharedExecutor(), quorumSize());
+    }
 
-        return ImmutableNetworkClientFactories.builder()
-                .acceptor(acceptorFactory::paxosAcceptorForClient)
-                .learner(learnerFactory::paxosLearnerForClient)
-                .addCloseables(acceptorFactory, learnerFactory)
-                .build();
+    @Value.Derived
+    @Override
+    public List<Closeable> closeables() {
```

Comment:
This can probably be auxiliary too, given it depends on acceptor and learner which we're already comparing